Sources/GroundState/py_variational_montecarlo.hpp

```diff
@@ -29,9 +29,84 @@ namespace py = pybind11;
 
 namespace netket {
 
+namespace detail {
+template <class T, int ExtraFlags>
+py::array_t<T, ExtraFlags> as_readonly(py::array_t<T, ExtraFlags> array) {
+  py::detail::array_proxy(array.ptr())->flags &=
+      ~py::detail::npy_api::NPY_ARRAY_WRITEABLE_;
+  return array;
+}
+}  // namespace detail
+
 void AddVariationalMonteCarloModule(py::module &m) {
   auto m_vmc = m.def_submodule("variational");
 
+  py::class_<Stats>(m_vmc, "Stats")
+      .def_readonly("mean", &Stats::mean)
+      .def_readonly("error_of_mean", &Stats::error_of_mean)
+      .def_readonly("variance", &Stats::variance)
+      .def_readonly("autocorrelation", &Stats::correlation)
```

Comment:
Calling this `autocorrelation` (or `correlation` in C++) is somewhat misleading, because this quantity is actually (an estimate of) the _autocorrelation time_ of the Markov chains. Even though this is a bit long, I'd recommend using this name, otherwise it is just confusing (since the real autocorrelation is a different quantity of different dimensions).

If we can write `error_of_mean`, then we can also write `autocorrelation_time`.

Reply:
this was called `tau_corr` before... just saying